### PR TITLE
Make dropdowns update correctly when you switch between images.

### PR DESCRIPTION
**Pull request by rachel-fenichel**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

https://github.com/google/blockly/issues/1047
and its dupes: https://github.com/google/blockly/issues/1232 and https://github.com/google/blockly/issues/1287

### Proposed Changes

Force a rerender of the parent block if the value of a dropdown changed, instead of only when the text changed.  The text of an image item in a dropdown is the alt-text.  If the developer didn't specify alt-text, the image wouldn't update.

### Reason for Changes

The previous behaviour was incorrect in a user-facing way.

### Test Coverage

Tested in the playground.

Tested on:
- [ ] Desktop:
  - [x] Chrome
  - [ ] Firefox
  - [ ] Safari
  - [ ] Opera
  - [ ] IE 10+
  - [ ] IE 11
  - [ ] EDGE

- [ ] Smartphone/Tablet/Chromebook (please complete the following information):

### CODE REVIEW

File: core/field_dropdown.js [hunk 1]
```diff
@@ -413,17 +413,20 @@ Blockly.FieldDropdown.prototype.setValue = function(newValue) {
       var content = options[i][0];
       if (typeof content == 'object') {
         this.imageJson_ = content;
-        this.setText(content.alt);
+        this.text_ = content.alt;
```
Comment (picklesrus) [inline core/field_dropdown.js]: Why did you change these away from setText?

Comment (rachel-fenichel) [reply core/field_dropdown.js]: Because if the text hasn't changed setText won't do the rerender, so I would need to rerender here.  But if the text had changed, setText would do a rerender and I wouldn't know (except by storing the old and new values here and comparing them), so forcing a rerender would cause an extraneous rendering pass.  I didn't want to change setText to return whether it updated, since that would be a change on Field.  So I'm just doing the actual effects of setText that I care about.

Comment (picklesrus) [reply core/field_dropdown.js]: Okay.  More evidence that we need to clean up fields and setText/setValue!  But, I agree that this bug is worth fixing and this is probably an okay work around for now.  It might cause an awkward conflict in scratch-blocks so we should watch for that.
